connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConnectorTest.java

```diff
@@ -85,7 +87,7 @@ public void testInitializeFailure() {
 
         replayAll();
 
-        WorkerConnector workerConnector = new WorkerConnector(CONNECTOR, connector, ctx, metrics, listener);
+        WorkerConnector workerConnector = new WorkerConnector(CONNECTOR, connector, ctx, metrics, listener, offsetStorageReader);
```

Comment:
So this existing test is a little strange because it does not instantiate a source or sink connector but merely relies upon the WorkerConnector for the most part has the same behavior regardless of whether the `Connector` instance is an instance of `SourceConnector` or `SinkConnector`. This PR changes this behavior, and I think we need to add a bit more testing here.

I'd suggest adding three more tests that simply verify the correct context type is passed to the `connector.initialize(...)` method, and that these tests verify the context is:
* a `ConnectorContext` instance if the connector is neither a source or a sink
* a `SinkConnectorContext` instance if the connector is a sink
* a `SourceConnectorContext` instance if the connector is a source

Reply:
Added more tests and changed existing tests to use source or sink and not an ambiguous connector.